### PR TITLE
HCAL: Modifications to the LUT and trigger key generation script

### PR DESCRIPTION
**Pull request by Michael-Krohn**
#### PR description:

Modifications to the HCAL LUTs and trigger key generation script, with the changes that are described in slide 3 [here](https://indico.cern.ch/event/1182049/contributions/4965979/attachments/2480788/4258655/O2OStatusUpdate_71522.pdf). This is needed for the new o2o system for updating online conditions in the HCAL. Also, removed ZDC from the LUT generation as there is an issue with the LUT generation of the ZDC that causes the scripts to fail and at the moment ZDC does not need LUTs.

#### PR validation:

A basic technical test was performed: runTheMatrix.py -l limited -i all --ibeos

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

This PR is not a backport. This PR updates the "service" scripts for LUTs and Trigger key generation, so the changes do not need to be backported to 12_4_X

### CODE REVIEW

File: CaloOnlineTools/HcalOnlineDb/src/HcalLutManager.cc [hunk 1]
```diff
@@ -1703,8 +1703,8 @@ int HcalLutManager::createLutXmlFiles_HBEFFromCoder_HOFromAscii_ZDC(std::string
     addLutMap(xml, masks);
   }
   //
-  const std::map<int, std::shared_ptr<LutXml>> _zdc_lut_xml = getZdcLutXml(_tag, split_by_crate);
-  addLutMap(xml, _zdc_lut_xml);
+  //  const std::map<int, std::shared_ptr<LutXml>> _zdc_lut_xml = getZdcLutXml(_tag, split_by_crate);
+  //  addLutMap(xml, _zdc_lut_xml);
```
Comment (tvami) [inline CaloOnlineTools/HcalOnlineDb/src/HcalLutManager.cc:1707]: Please remove commented code

**Comment by tvami** [comment]
> If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:
> This PR is not a backport.

Did you read the second part of the question? It's certainly not answered... Based on the desc I think you need this in 12_4_X, correct?

**Comment by abdoulline** [comment]
@tvami  @Michael-Krohn 
I believe this "service" script (for LUTs + Trigger Key  generation) update doesn't need to be backported.

**Comment by tvami** [comment]
Thanks @abdoulline good to know it's not needed in 12_4_X, that makes life easier... @Michael-Krohn please update the PR desc with this info. Thanks!

**Comment by tvami** [comment]
@Michael-Krohn please squash the commits

**Comment by perrotta** [comment]
I suspect that something got wrong in the squash of the commits...

**Comment by Michael-Krohn** [comment]
Yes, sorry. Somehow while squashing commits I accidentally deleted my branch. I recreated my branch and made a new PR https://github.com/cms-sw/cmssw/pull/39154

**Comment by tvami** [comment]
-db
 * new PR was made

**Pull request merged**